cmake/Modules/EttercapLibCheck.cmake

```diff
@@ -35,18 +35,30 @@ if(ENABLE_GTK)
         message(STATUS "No GTK_BUILD_TYPE defined, default is GTK2")
         set(GTK_BUILD_TYPE GTK2 CACHE STRING 
         "Choose the type of gtk build, options are: ${VALID_GTK_TYPES}." FORCE)
+    else(NOT DEFINED GTK_BUILD_TYPE)
+        set(GTK_BUILD_TYPE ${GTK_BUILD_TYPE} CACHE STRING
+        "Choose the type of gtk build, options are: ${VALID_GTK_TYPES}." FORCE)
     endif(NOT DEFINED GTK_BUILD_TYPE)
     LIST(FIND VALID_GTK_TYPES ${GTK_BUILD_TYPE} contains_valid)
     if(contains_valid EQUAL -1)
         message(FATAL_ERROR "Unknown GTK_BUILD_TYPE: '${GTK_BUILD_TYPE}'. Valid options are: ${VALID_GTK_TYPES}")
     endif()
     UNSET(contains_valid)
     if(GTK_BUILD_TYPE STREQUAL GTK3)
+        set(GTK3_FIND_VERSION 1)
         find_package(GTK3 REQUIRED)
         if(NOT GTK3_FOUND)
             message(FATAL_ERROR "You choose to build against GTK3, please install it, or build against GTK2")
         endif(NOT GTK3_FOUND)
-        set(HAVE_GTK3 1)
+        if (GTK3_MINOR_VERSION LESS 12)
+            message(STATUS "Your version of GTK3 (${GTK3_VERSION}) is not sufficient for full GTK3 support.")
+            message(STATUS "Full support requires >= 3.12. Building in GTK3 compatibility mode.")
+            set(GTK_BUILD_SUBTYPE "GTK3COMPAT")
```

Comment:
the GTK_BUILD_SUBTYPE variable seems unused...

Reply:
nevermind

Reply:
Did you try this Gtk3 on Windows too? If so, where/how did you get that dev-package?

Reply:
No. not yet.

Have you tried the two projects I recommended to you? [gvsbuild](https://github.com/wingtk/gvsbuild) and Microsoft's [vcpkg](https://github.com/Microsoft/vcpkg)

Reply:
These have no prebuilt .zip of Gtk > 3.12 AFAICS. Have to build yourself. Yuk!
Those Gnome nerds seems to loath Windows users. We'll be better of with wxWidgets and Qt5.

Reply:
> Have to build yourself. Yuk!

Yep. But it should be pretty straightforward. Especially with vcpkg.
The only prebuilds I know is the one that comes with msys2, and it's not msvc-compatiple.
I suppose you could also grab gtk3 prebuilds from appveyor's builder. You know, write a dummy yalm that copies zips and uploads them. 

>We'll be better of with wxWidgets and Qt5.

I have bad experiences with using Qt5. And the hell is it so big?